Core/Src/main.c

```diff
@@ -87,17 +97,26 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
-  /* USER CODE BEGIN 2 */
+  MX_SPI1_Init();
+  MX_TIM1_Init();
 
+  /* USER CODE BEGIN 2 */
+  HAL_TIM_PWM_Init(&htim1); //start PWM
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
```

Comment:
Don't forget to bring the CS (chip select) line low before you transmit and receive and high again in between communication. The ADC will not listen unless CS is Low.
Have a look at the HAL_GPIO stuff on writing to a pin.